unstructured_ingest/v2/processes/connectors/mongodb.py

```diff
@@ -53,6 +63,206 @@ class MongoDBUploadStagerConfig(UploadStagerConfig):
     pass
 
 
+class MongoDBIndexerConfig(IndexerConfig):
+    batch_size: int = Field(default=100, description="Number of records per batch")
+
+
+class MongoDBDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class MongoDBIndexer(Indexer):
+    connection_config: MongoDBConnectionConfig
+    index_config: MongoDBIndexerConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    def precheck(self) -> None:
+        """Validates the connection to the MongoDB server."""
+        try:
+            client = self.create_client()
+            client.admin.command("ping")
+        except Exception as e:
+            logger.error(f"Failed to validate connection: {e}", exc_info=True)
+            raise SourceConnectionError(f"Failed to validate connection: {e}")
+
+    @requires_dependencies(["pymongo"], extras="mongodb")
+    def create_client(self) -> "MongoClient":
+        from pymongo import MongoClient
+        from pymongo.driver_info import DriverInfo
+        from pymongo.server_api import ServerApi
+
+        access_config = self.connection_config.access_config.get_secret_value()
+
+        if access_config.uri:
+            return MongoClient(
+                access_config.uri,
+                server_api=ServerApi(version=SERVER_API_VERSION),
+                driver=DriverInfo(name="unstructured", version=unstructured_version),
+            )
+        else:
+            return MongoClient(
+                host=self.connection_config.host,
+                port=self.connection_config.port,
+                server_api=ServerApi(version=SERVER_API_VERSION),
+            )
+
+    @requires_dependencies(["pymongo"], extras="mongodb")
+    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
+        """Generates FileData objects for each document in the MongoDB collection."""
+        client = self.create_client()
+        database = client[self.connection_config.database]
+        collection = database[self.connection_config.collection]
+
+        # Get list of document IDs
+        ids = collection.distinct("_id")
+        batch_size = self.index_config.batch_size if self.index_config else 100
+        id_batches = [ids[i : i + batch_size] for i in range(0, len(ids), batch_size)]
+
+        for id_batch in id_batches:
+            batch_id = str(hash(frozenset(id_batch)) + sys.maxsize + 1)
+
+            metadata = FileDataSourceMetadata(
+                date_processed=str(time()),
+                record_locator={
+                    "database": self.connection_config.database,
+                    "collection": self.connection_config.collection,
+                },
+            )
+
+            file_data = FileData(
+                identifier=batch_id,
```

Comment:
The file data type should be `batch`. It defaults to `file`: [file_data.py#L41](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/v2/interfaces/file_data.py#L41)
```suggestion
                identifier=batch_id,
                doc_type="batch",
```